src/test/ui/pattern/issue-82290.rs

```diff
@@ -1,7 +1,9 @@
+// check-pass
+
 #![feature(let_chains)] //~ WARN the feature `let_chains` is incomplete
 
 fn main() {
-    if true && let x = 1 { //~ ERROR `let` expressions are not supported here
+    if true && let x = 1 { //~ WARN irrefutable `let` pattern
```

Comment:
Originally created because of an ICE -> https://github.com/rust-lang/rust/issues/82290